source/io/TlsOptions.cpp

```diff
@@ -283,6 +284,13 @@ namespace Aws
 
             TlsConnectionOptions TlsContext::NewConnectionOptions() const noexcept
             {
+                if (!*this)
```

Comment:
This gives me...feelings, but I get why it's here. Syntactically, it's a bit... obtuse. Is there not an IsValid() or similar we could call instead?

Reply:
Negative, the class just overloads the operator right now.  I can certainly add one though!